### PR TITLE
docs(reference/typescript): remove type provider from typebox example

### PR DESCRIPTION
**Pull request by jscheffner**
I made some small adjustments to the Typescript docs:

* I removed the type provider from the Typebox example. The example uses both a type provider and a generic which makes one of them obsolete. Also to my understanding this part of the docs is supposed to show how to use those libraries without type providers. There's a separate documentation of the type providers.
* I rephrased some sentences, just minor things.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by climba03003 — APPROVED** [review]
Make sense to remove the `.withTypeProvider`.
It is duplicated.

File: docs/Reference/TypeScript.md [hunk 1]
```diff
@@ -210,24 +210,23 @@ And a `zod` wrapper by a third party called [`fastify-type-provider-zod`](https:
 They simplify schema validation setup and you can read more about them in [Type
 Providers](./Type-Providers.md) page.
 
-Below is how to setup schema validation using _vanilla_ `typebox` and
-`json-schema-to-ts` packages.
+Below is how to setup schema validation using the `typebox`, `json-schema-to-typescript` and
```
Comment (Fdawgs) [inline docs/Reference/TypeScript.md:213]:
```suggestion
Below is how to setup schema validation using the `typebox`, `json-schema-to-typescript`, and
```

Missing Oxford comma.

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by mcollina — CHANGES_REQUESTED** [review]
I don't understand why the type provider has been removed. Can you clarify?

**Comment by jscheffner** [comment]
> I don't understand why the type provider has been removed. Can you clarify?

Yes I can. There are two mechanisms to get the handler typed:

* You can use the route generic
* You can use a type provider so you don't have to use the generics

But it doesn't help to use both. So in the example there should either be no type provider or no generic. There's a link to the type provider docs just above the examples. There you can find examples using the type providers. Also it says "Below is how to setup schema validation using vanilla typebox and json-schema-to-ts packages." So I think the purpose of the examples was to show how to use the libraries with generics.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Fdawgs — APPROVED** [review]

**Review by gurgunday — APPROVED** [review]

**Pull request merged**